### PR TITLE
feat: Use autogenerated JSON schema

### PR DESCRIPTION
**Pull request by candiduslynx**
Follow-up for #13876
Blocked by:
- [x] https://github.com/invopop/jsonschema/pull/103
- [x] https://github.com/invopop/jsonschema/pull/105
- [x] https://github.com/cloudquery/codegen/pull/29
- [x] https://github.com/cloudquery/plugin-sdk/pull/1254

### CODE REVIEW

File: plugins/source/gcp/go.mod [hunk 1]
```diff
@@ -68,12 +71,17 @@ require (
 // TODO: remove once all updates are merged
 replace github.com/apache/arrow/go/v14 => github.com/cloudquery/arrow/go/v14 v14.0.0-20230916001126-ffb7089245ac
 
+// https://github.com/invopop/jsonschema/pull/103
+replace github.com/invopop/jsonschema => github.com/cloudquery/jsonschema v0.0.0-20230926171440-9c0d982c4f4f
```
Comment (candiduslynx) [inline plugins/source/gcp/go.mod:75]: waiting for https://github.com/invopop/jsonschema/pull/103 & bump in codegen

File: plugins/source/gcp/go.mod [hunk 2]
```diff
@@ -68,12 +70,20 @@ require (
...
+
+// https://github.com/cloudquery/codegen/pull/29
+replace github.com/cloudquery/codegen => github.com/cloudquery/codegen v0.3.3-0.20230928065241-2c7e0070ce1e
```
Comment (candiduslynx) [inline plugins/source/gcp/go.mod:77]: waiting for https://github.com/invopop/jsonschema/pull/103 & https://github.com/cloudquery/codegen/pull/29

File: plugins/source/gcp/client/spec/schema.json [hunk 3]
```diff
@@ -1,159 +1,127 @@
...
-          "minLength": 1
-        }
-      },
       "additionalProperties": false,
-      "required": [
-        "target_principal"
-      ]
+      "type": "object"
+    },
+    "Strategy": {
```
Comment (candiduslynx) [inline plugins/source/gcp/client/spec/schema.json:116]: Blocked by https://github.com/cloudquery/plugin-sdk/pull/1254

File: plugins/source/gcp/client/spec/schema.json [hunk 4]
```diff
@@ -1,159 +1,127 @@
...
           "items": {
             "type": "string",
             "minLength": 1
-          }
+          },
+          "type": "array"
         },
         "folder_ids": {
-          "title": "Folder IDs",
-          "description": "Location of the projects to sync from. Mutually exclusive with project_filter",
```
Comment (hermanschaaf) [inline plugins/source/gcp/client/spec/schema.json:20]: What happened to all the descriptions?

Comment (candiduslynx) [reply plugins/source/gcp/client/spec/schema.json:20]: I removed them (they can be added via `jsonschema_description` tag, but the code lines would look too long).

I guess we can experiment with the descriptions & generating docs from the schema later, but for now, the descriptions don't give any additional info to the validation errors, so I suppose it's OK to omit them.

**Comment by hermanschaaf** [comment]
Could we split the GCP SDK update into a separate PR so that it's separate from the JSONSchema update and shows as a different entry in the changelog?

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**